src/Dscp/DB/SQLite/Instances.hs

```diff
@@ -28,16 +32,21 @@ instance FromField Course            where fromField f = Course            <$> f
 instance FromField Grade             where fromField f = UnsafeGrade       <$> fromField f
 instance FromField AssignmentType    where fromField f = Codec.deserialise <$> fromField f
 instance FromField SubmissionWitness where fromField f = Codec.deserialise <$> fromField f
+instance FromField ATGDelta          where fromField f = Codec.deserialise <$> fromField f
 
 instance ToField   (Hash a)          where toField = toField . Codec.serialise
 instance ToField   (Signature a)     where toField = toField . Codec.serialise
+instance ToField (MerkleSignature a) where toField = toField . Codec.serialise
+instance Serialise a =>
+         ToField   (MerkleTree a)    where toField = toField . Codec.serialise
```

Comment:
Won't work as expected - `MerkleTree a` has actual values of `a` inside it (which in our case are actual transactions). We definitely don't want to serialise transactions themselves into this field.
The simplest way to do it right, imho, is to have a `newtype EmptyMerkleTree a = Empty (MerkleTree ())` with functions like `getEmptyMerkleTree :: MerkleTree a -> EmptyMerkleTree a` and `fillEmptyMerkleTree :: Map LeafIndex a -> EmptyMerkleTree a -> MerkleProof a`, and serialise this newtype into SQLite instead, making it from a full Merkle tree on block creation and combining it with fetched transactions to get a proof after fetching.

Reply:
Done.